url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/repl/ExtensionRepositoryBuilder.java

```diff
@@ -0,0 +1,397 @@
+                }
+            }
+
+            if(lLibInstance == null && lLibraryType == ScriptyLibraryType.INSTANCE)
+            {
+                throw new ExtensionException(String.format("Library '%s' was marked as being of type INSTANCE, but it was impossible to create an instance.\nMaybe you should create an instance manually and provide the instance.", lClass.getName()));
+            }
+
+            addLibrary(lLibraryName, lLibInstance, lClass);
+        }
+    }
+    
+    private void registerClassArgList(Class aClass, ScriptyStdArgList aStdLst, Map<String, ArgListBuilderUtil.Tuple<IArgList, Map<String, IArgMapping>>> aNamedLists)
+    throws ExtensionException
+    {
+        final String lName = aStdLst.name();
+        if(lName.length() == 0)
+        {
+            final String lMsg = String.format("Class '%s' contains an unnamed standard argument list specification. An argument list specification associated with a command library should always have a name.", aClass.getName());
+            throw new ExtensionException(lMsg);
+        }
+
+        try
+        {
+            // Build it.
+            final ArgListBuilderUtil.Tuple<IArgList, Map<String, IArgMapping>> lTuple = ArgListBuilderUtil.buildArgList(aStdLst);
+            // Remember it for references if the argument list spec has a name that is.
+            aNamedLists.put(lName, lTuple);
+        }
+        catch (ArgSpecException e)
+        {
+            final String lMsg = String.format("Class '%s' contains a standard argument list specification '%s' with errors.\n%s", aClass.getName(), aStdLst.name(), e.getMessage());
+            throw new ExtensionException(lMsg, e);
+        }
+    }
+
+    private void registerClassArgList(Class aClass, ScriptyVarArgList aVarLst, Map<String, ArgListBuilderUtil.Tuple<IArgList, Map<String, IArgMapping>>> aNamedLists)
+    throws ExtensionException
+    {
+        final String lName = aVarLst.name();
+        if(lName.length() == 0)
+        {
+            final String lMsg = String.format("Class '%s' contains an unnamed variable argument list specification. An argument list specification associated with a command library should always have a name.", aClass.getName());
+            throw new ExtensionException(lMsg);
+        }
+
+        try
+        {
+            // Build it.
+            final ArgListBuilderUtil.Tuple<IArgList, Map<String, IArgMapping>> lTuple = ArgListBuilderUtil.buildArgList(aVarLst);
+            // Remember it for references if the argument list spec has a name that is.
+            aNamedLists.put(lName, lTuple);
+        }
+        catch (ArgSpecException e)
+        {
+            final String lMsg = String.format("Class '%s' contains a variable argument list specification '%s' with errors.\n%s", aClass.getName(), aVarLst.name(), e.getMessage());
+            throw new ExtensionException(lMsg, e);
+        }
+    }
+
+    private void addLibrary(String aLibName, Object aLibInstance, Class aClass)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 38. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 161-197 into a separate method.